django/db/models/query.py

```diff
@@ -1051,10 +1058,20 @@ def _batched_insert(self, objs, fields, batch_size):
             return
         ops = connections[self.db].ops
         batch_size = (batch_size or max(ops.bulk_batch_size(fields, objs), 1))
+        ret = []
         for batch in [objs[i:i + batch_size]
                       for i in range(0, len(objs), batch_size)]:
-            self.model._base_manager._insert(batch, fields=fields,
-                                             using=self.db)
+            if connections[self.db].features.can_return_ids_from_bulk_insert:
+                if len(objs) > 1:
+                    ret.extend(self.model._base_manager._insert(batch, fields=fields,
```

Comment:
I wouldn't repeat `self.model._base_manager._insert(batch, fields=fields, ...` but just vary the extend/append action based on the number. Are both branches tested?